common/web/lm-worker/src/main/correction/context-tracker.ts

```diff
@@ -391,100 +374,94 @@ export class ContextTracker extends CircularArray<TrackedContextState> {
-      *   - That is, no "reasonable" keystroke would emit enough code points to 'bump' two words simultaneously.
-      *   - ... This one may need to be loosened a bit... but it should be enough for initial correction testing as-is.
-      * - Assumption:  one keystroke may only cause a single token to be appended to the context
-      *   - That is, no "reasonable" keystroke would emit a Transform adding two separate word tokens
-      *     - For languages using whitespace to word-break, said keystroke would have to include said whitespace to break the assumption.
-      */
-
-    function maintainLastToken() {
-      if(isWhitespace && editPath[tailIndex] == 'match') {
-        /*
-          We can land here if there are multiple whitespaces in a row.
-          There's already an implied whitespace to the left, so we conceptually
-          merge the new whitespace with that one.
-        */
-        return;
-      } else if(isBackspace) {
-        // Consider backspace entry for this case?
-        state.replaceTailForBackspace(finalToken, primaryInput.id);
-      } else {
-        state.updateTail(primaryInput ? transformDistribution : null, finalToken);
-      }
-    }
 
-    // If there is/was more than one context token available...
-    if(editPath.length > 1) {
-      // We're removing a context token, but at least one remains.
-      if(poppedHead) {
-        state.popHead();
-      }
+    // Reset priorEdit for the end-of-context updating loop.
+    priorEdit = undefined;
 
-      // We're adding an additional context token.
-      if(pushedTail) {
-        const tokenizedTail = tokenizedContext[tokenizedContext.length - 1];
-        /*
-          * Common-case:  most transforms that trigger this case are from pure-whitespace Transforms.  MOST.
-          *
-          * Less-common, but noteworthy:  some wordbreaks may occur without whitespace.  Example:
-          * `"o` => ['"', 'o'].  Make sure to double-check against `tokenizedContext`!
-          */
-        let pushedToken = new TrackedContextToken();
-        pushedToken.raw = tokenizedTail;
-
-        if(isWhitespace || !primaryInput) {
-          state.pushWhitespaceToTail(transformDistribution ?? []);
-          // Continuing the earlier assumption, that 'pure-whitespace Transform' does not emit any initial characters
-          // for the new word (token), so the input keystrokes do not correspond to the new text token.
-          pushedToken.transformDistributions = [];
-        } else {
-          state.pushWhitespaceToTail();
-          // Assumption: Since we only allow one-transform-at-a-time changes between states, we shouldn't be missing
-          // any metadata used to construct the new context state token.
-          pushedToken.transformDistributions = transformDistribution ? [transformDistribution] : [];
-        }
+    // Now to update the end of the context window.
+    for(let i = lastMatch+1; i < editPath.length; i++) {
```

Comment:
Note:  as is, this change _does_ break maintenance of the correction-search oriented searchState field.  This issue is corrected in the followup (#11997), as use of the method added in #11998 (not integrated here) is needed to accomplish this in full.